libcloud/compute/drivers/libvirt_driver.py

```diff
@@ -82,16 +82,18 @@ def __init__(self, uri, key=None, secret=None):
         self._uri = uri
         self._key = key
         self._secret = secret
-        try:
-            self.connection = libvirt.open(uri)
-        except libvirt.libvirtError:
-            if key is None or secret is None:
+        if '+tcp' in self._uri:
+            if key is None and secret is None:
                 raise RuntimeError('The remote Libvirt instance requires ' +
                                    'authentication, please set \'key\' and ' +
                                    '\'secret\' parameters')
             auth = [[libvirt.VIR_CRED_AUTHNAME, libvirt.VIR_CRED_PASSPHRASE],
                     self._cred_callback, None]
             self.connection = libvirt.openAuth(uri, auth, 0)
+        else:
+            self.connection = libvirt.open(uri)
```

Comment:
Hm, so `open` simply returns `None` if it's unable to establish a connection?

If so, that's a bit unfriendly and I guess if you want to debug it you need to set `LIBVIR_DEBUG` environment variable, or is there any other, more user-friendly way to do it?